src/teacher.py

```diff
@@ -51,7 +51,11 @@ def _answer_Demonstration_query(
         except (ApproachTimeout, ApproachFailure):
             return DemonstrationResponse(query, teacher_traj=None)
 
-        teacher_traj, _, goal_reached = utils.run_policy_on_task(
+        traj, _, goal_reached = utils.run_policy_on_task(
             policy, task, self._simulator, CFG.max_num_steps_option_rollout)
         assert goal_reached
+        teacher_traj = LowLevelTrajectory(traj.states,
+                                          traj.actions,
+                                          _is_demo=True,
```

Comment:
to answer your comment on the PR, i think this is fine. anyone bold enough to use DemonstrationQuery/Response must be willing to accept the semantic consequences.